app/core/logging.py

```
@@ -0,0 +1,222 @@
+        structlog.processors.StackInfoRenderer(),
+    ]
+    
+    # Add sensitive data masker
+    processors.append(SensitiveDataMasker(log_config.mask_fields))
+    
+    # Environment-specific processors
+    if environment == "development":
+        processors.append(structlog.processors.format_exc_info)
+    else:
+        # In production, we want exception info but not full stack traces
+        def format_exception_short(_, __, event_dict):
```

Ruff (0.11.9)

Missing return type annotation for private function `format_exception_short`

(ANN202)